### PR TITLE
MIGENG-170: Filtering Workload Inventory Report - UI Changes

### PR DESCRIPTION
**Pull request by carlosthe19916**
Work in progress.
This PR depends on https://github.com/project-xavier/xavier-ui/pull/32

https://jira.coreos.com/browse/MIGENG-170

### CODE REVIEW

**Comment by carlosthe19916** [comment]
PR ready for review.
Latest changes added new filters as you can see in the next screenshots:

Screenshot from 2019-08-08 11-43-28
Screenshot from 2019-08-08 11-43-48
Screenshot from 2019-08-08 11-44-13

File: src/pages/ReportView/WorkloadInventory/WorkloadInventory.tsx [hunk 1]
```diff
@@ -86,8 +100,52 @@ interface State {
...
+    [FilterTypeKeyEnum.EFFORT, "Effort"],
+    [FilterTypeKeyEnum.RECOMMENDED_TARGETS_IMS, "Rec. Targets"],
+    [FilterTypeKeyEnum.FLAGS_IMS, "flags IMS"],
+]);
+
+const primaryFilters = [
+    { name: 'Provider', value: FilterTypeKeyEnum.PROVIDER },
+    { name: 'Datacenter', value: FilterTypeKeyEnum.DATACENTER },
+    { name: 'Cluster', value: FilterTypeKeyEnum.CLUSTER },
+    { name: 'Vm name', value: FilterTypeKeyEnum.VM_NAME },
```
Comment (mrizzi) [inline src/pages/ReportView/WorkloadInventory/WorkloadInventory.tsx]:
```suggestion
    { name: 'VM name', value: FilterTypeKeyEnum.VM_NAME },
```

File: src/pages/ReportView/WorkloadInventory/WorkloadInventory.tsx [hunk 2]
```diff
@@ -86,8 +100,52 @@ interface State {
...
+    { name: 'Workload', value: FilterTypeKeyEnum.WORKLOAD },
+    { name: 'Os name', value: FilterTypeKeyEnum.OS_NAME },
```
Comment (mrizzi) [inline src/pages/ReportView/WorkloadInventory/WorkloadInventory.tsx]:
```suggestion
    { name: 'OS type', value: FilterTypeKeyEnum.OS_NAME },
```

File: src/pages/ReportView/WorkloadInventory/WorkloadInventory.tsx (hunk 1 above)
Comment (carlosthe19916) [reply src/pages/ReportView/WorkloadInventory/WorkloadInventory.tsx]: @mrizzi Thanks Marco.

File: src/pages/ReportView/WorkloadInventory/WorkloadInventory.tsx (hunk 2 above)
Comment (carlosthe19916) [reply src/pages/ReportView/WorkloadInventory/WorkloadInventory.tsx]: @mrizzi Thanks Marco.

**Comment by carlosthe19916** [comment]
> Some initial comments on some filters' labels and there's also the issue i reported about the page to crash when a filter with no values is selected (`Flags IMS` in my case)

@mrizzi it has been fixed in my last commit. Now, when there is no value available for the dropdown, I will show a message. See the screenshot below:

Screenshot from 2019-08-30 17-00-51

**Comment by carlosthe19916** [comment]
> Some more feedback:
> 
> 1. change `Select a filter...` label to be _`Filter by...`_ with _italic_ style when the page is loaded for the first time (see screenshot)
>    Screenshot from 2019-08-30 17-20-08
> 2. once the filter is select, e.g. `Workload` the `Filter by Workload` has to be grey (not black) just like `Select a filter...` label is and it should be _italiac_ as well
> 3. the checkboxes are different in the UI (see next screeshot) from the mockups where they are light blue when selected: is this something we can change using standard PF/React components?
>    Screenshot from 2019-08-30 17-27-27
> 4. when i click `Clear filters` i would have expect the filter box to go back to the first state with `Filter` and `Filter by` labels and not to have the last select filter still selected

- **Change Select a filter... label to be Filter by...**: Done
- Even though the mockups describe the dropdown filters using ITALIC, the components from https://www.patternfly.org/v4/documentation/react/components/select/#checkbox-select-input does not use Italic font. I suggest using the default config unless UX team indicates the contrary.
- **once the filter is select, e.g. Workload the Filter by Workload has to be grey (not black) just like Select a filter..**: Again the default config from https://www.patternfly.org/v4/documentation/react/components/select/#checkbox-select-input does not include gray color in the first element.
- **when i click Clear filters i would have expect the filter box to go back to the first state with Filter and Filter by labels and not to have the last select filter still selected**: done

**Pull request merged**